lib/bsdlib/nrf91_sockets.c

```diff
@@ -478,24 +478,45 @@ static int nrf91_socket_offload_accept(int sd, struct sockaddr *addr,
 				       socklen_t *addrlen)
 {
 	int retval;
+	struct nrf_sockaddr *nrf_addr_ptr = NULL;
+	nrf_socklen_t *nrf_addrlen_ptr = NULL;
+	/* Use `struct nrf_sockaddr_in6` to fit both, IPv4 and IPv6 */
+	struct nrf_sockaddr_in6 nrf_addr;
+	nrf_socklen_t nrf_addrlen;
+
+	if ((addr != NULL) && (addrlen != NULL)) {
```

Comment:
This line should be the parameter check in the very beginning. Return error if any of them is NULL. Saving all the re-checks below.